pkg/builtin/builtin.go

```diff
@@ -489,6 +489,12 @@ func SysRead(_ context.Context, _ []string, input string, _ chan<- string) (stri
 	if len(data) == 0 {
 		return fmt.Sprintf("The file %s has no contents", params.Filename), nil
 	}
+
+	// Assume the file is not text if it contains a null byte
+	if bytes.Contains(data, []byte{0}) {
```

Comment:
nit: `bytes.IndexByte` is probably slightly more performant.

Reply:
Fixed